pulsar-broker/src/main/java/org/apache/pulsar/broker/PulsarService.java

```diff
@@ -446,7 +441,16 @@ public void close() throws PulsarServerException {
 
             List<CompletableFuture<Void>> asyncCloseFutures = new ArrayList<>();
             if (this.brokerService != null) {
-                asyncCloseFutures.add(this.brokerService.closeAsync());
+                CompletableFuture<Void> brokerCloseFuture = this.brokerService.closeAsync();
+                if (this.transactionMetadataStoreService != null) {
+                    asyncCloseFutures.add(brokerCloseFuture.thenAccept(__ -> {
```

Comment:
should this be "whenComplete" o "handle" ? otherwise in case of failure we are not executing this code

Reply:
thanks, good catch.

Reply:
@eolivelli PTAL